Penteract/Assets/Scripts/HUDController.cpp

```diff
@@ -340,11 +362,88 @@ void HUDController::UpdateFangCooldowns(GameObject* fangSkillCanvas) {
 				}
 			}
 		}
+
 		++skill;
 	}
 }
 
-void HUDController::UpdateOnimaruCooldowns(GameObject* onimaruSkillCanvas) {
+
+
+void HUDController::PlayCoolDownEffect(AbilityRefreshEffect* ef, Cooldowns cooldown) {
+	if (ef != nullptr) {
+		ef->Play();
+		abilityCoolDownsRetreived[cooldown] = true;
+	}
+}
+
+
+void HUDController::AbilityCoolDownEffectCheck(Cooldowns cooldown, GameObject* canvas) {
+	if (!abilityCoolDownsRetreived[cooldown]) {
+		if (canvas) {
+			if (cooldowns[cooldown] == 1.0f) {
+
+				AbilityRefreshEffect* ef = nullptr;
+				switch (cooldown) {
+				case Cooldowns::FANG_SKILL_1:
+					if (canvas->GetChildren().size() > 0) {
+						if (canvas->GetChildren()[0]->GetChildren().size() > 4) {
+
+							ef = GET_SCRIPT(canvas->GetChildren()[0]->GetChildren()[4], AbilityRefreshEffect);
+						}
+					}
+					break;
+				case Cooldowns::FANG_SKILL_2:
+					if (canvas->GetChildren().size() > 1) {
+						if (canvas->GetChildren()[1]->GetChildren().size() > 4) {
+							ef = GET_SCRIPT(canvas->GetChildren()[1]->GetChildren()[4], AbilityRefreshEffect);
+						}
+					}
+					break;
+				case Cooldowns::FANG_SKILL_3:
+					if (canvas->GetChildren().size() > 2) {
+						if (canvas->GetChildren()[2]->GetChildren().size() > 4) {
+							ef = GET_SCRIPT(canvas->GetChildren()[2]->GetChildren()[4], AbilityRefreshEffect);
+						}
+					}
+					break;
+				case Cooldowns::ONIMARU_SKILL_1:
+					if (canvas->GetChildren().size() > 0) {
+						if (canvas->GetChildren()[0]->GetChildren().size() > 4) {
+							AbilityRefreshEffect* ef = GET_SCRIPT(canvas->GetChildren()[0]->GetChildren()[4], AbilityRefreshEffect);
+							ef->Play();
+						}
+					}
+					break;
+				case Cooldowns::ONIMARU_SKILL_2:
+					if (canvas->GetChildren().size() > 0) {
+						if (canvas->GetChildren()[1]->GetChildren().size() > 4) {
+							ef = GET_SCRIPT(canvas->GetChildren()[1]->GetChildren()[4], AbilityRefreshEffect);
+						}
+					}
+					break;
+				case Cooldowns::ONIMARU_SKILL_3:
+					if (canvas->GetChildren().size() > 0) {
+						if (canvas->GetChildren()[2]->GetChildren().size() > 4) {
+							ef = GET_SCRIPT(canvas->GetChildren()[2]->GetChildren()[4], AbilityRefreshEffect);
+						}
+					}
+					break;
+				default:
+					if (canvas->GetChildren().size() > 4) {
+						ef = GET_SCRIPT(canvas->GetChildren()[4], AbilityRefreshEffect);
+					}
+					break;
+				}
+
+				PlayCoolDownEffect(ef, cooldown);
+			}
+		}
+	}
+}
+
+
+
+void HUDController::UpdateOnimaruCooldowns(GameObject* onimaruSkillCanvas, bool isMain) {
```

Comment:
isn't this the same function as Fang? Can't it be merged so you don't have two separate functions that appear to do the same?

Reply:
We may need to have a meet about this one.